pkg/accesscontrol/access_store.go

```diff
@@ -73,16 +86,34 @@ func (l *AccessStore) PurgeUserData(id string) {
 func (l *AccessStore) CacheKey(user user.Info) string {
 	d := sha256.New()
 
-	l.users.addRolesToHash(d, user.GetName())
-
 	groupBase := user.GetGroups()
-	groups := make([]string, 0, len(groupBase))
+	groups := make([]string, len(groupBase))
 	copy(groups, groupBase)
-
 	sort.Strings(groups)
-	for _, group := range user.GetGroups() {
-		l.groups.addRolesToHash(d, group)
+
+	l.addRolesToHash(d, user.GetName(), l.usersPolicyRules)
+	for _, group := range groups {
+		l.addRolesToHash(d, group, l.groupsPolicyRules)
 	}
 
 	return hex.EncodeToString(d.Sum(nil))
 }
+
+var null = []byte{'\x00'}
```

Comment:
This could be simplified to `[]byte{0}`

Reply:
Good point, thanks! In fact, writing this null does not seem to be even needed, so I'll just remove it.